core/src/main/java/com/datastax/oss/driver/internal/core/cql/Conversions.java

```diff
@@ -284,13 +289,39 @@ static AsyncResultSet toResultSet(
     }
   }
 
+  private static ColumnDefinitions getResultDefinitions(
+      Rows rows, Statement<?> statement, InternalDriverContext context) {
+    RowsMetadata rowsMetadata = rows.getMetadata();
+    if (rowsMetadata.columnSpecs.isEmpty()) {
+      // If the response has no metadata, it means the request had SKIP_METADATA set, the driver
+      // only ever does that for bound statements.
+      BoundStatement boundStatement = (BoundStatement) statement;
```

Comment:
I noticed this causes problems for some tests, but it appears its caused by simulacron not returning metadata for queries that weren't primed.  Instead it should probably return a `Void` result, i'll make that update.

Reply:
Oh, apparently simulacron already does this, it's just that our tests are priming `noRows()` instead of `void`.  Will fix, although I do wonder if in general it should throw some result metadata on there to be considered a valid select response.

Reply:
Updated simulacron, when noRows is specified we fake out some result metadata to match C* behavior.